### ISSUE TITLE
[Feature] Make it easier to trigger sync / sync on resume

### ISSUE THREAD

**Issue by mueller-ma** [TechbeeAT/jtxBoard#441]
**Is your feature request related to a problem? Please describe.**
I have a few tasks that are synced with davx5 and shared on the server with other people. Before I work on a task I want to make sure the task hasn't been done, so I trigger a manual sync. 

**Describe the solution you'd like**

- Implement a swipe-to-refresh layout that triggers a sync by using https://developer.android.com/jetpack/androidx/releases/swiperefreshlayout
- Sync when opening the app. This might be controlled by a setting.

**Additional context**
I can change the sync interval in davx5 to 15 minutes, but this will cause higher battery and data usage.

**Comment by patrickunterwegs** [comment]
Hey, thanks 👍
Swipe to refresh was there in jtx Board 1.x, actually people were complaining that they didn't wantit because it just triggered unnecessary syncs... 
But we can check it!.
I think the library that you mentioned is only for XML layouts, with the lazy list in compose this might get a bit more complicated.

But I have found this to check out:
https://developer.android.com/reference/kotlin/androidx/compose/material/pullrefresh/package-summary

About refreshing/synchronizing on App start or resume - I think it's a good idea :)

**Comment by rogercreagh** [comment]
Rather than swipe to full refresh would it make more sense to trigger a refresh of just the one item from the server when switching to edit view. It would mean a slight delay in opening the edit screen - perhaps a message saying something like "hang on a moment, just checking server version has not changed" to cover the delay.

I think it does sync when opening the app doesn't it? It certainly seems to always have latest remote changes, but maybe that's Davx5 doing its background thing.

**Comment by patrickunterwegs** [comment]
Currently there is no sync triggered on app start, that would be an easy thing to add.

I have also checked the swipe to refresh thing, this also seems to be easy: 
https://medium.com/google-developer-experts/effortlessly-add-pull-to-refresh-to-your-android-app-with-jetpack-compose-4c8b218a9beb

The only stupid thing is that I have to add the Material2 dependency once again...
For me it's important that such a swipe to refresh only appears when I'm really at the top of the list and then I want to keep scrolling.

Refreshing just one item is currently not possible with DAVx5... And refreshing the whole account could take more than a few seconds, so currently I wouldn't consider this as an option...

Basically those are the ways when a sync is triggered: 
- when saving a entry - here the sync framework is notified about a change, the sync might be delayed, this is up to the sync framework when it wants to sync
- when triggering a sync manually (from the top menu in the list, from the bottom icon in the detail view, from the sync screen) - here the sync framework is notified, but with an option that forces the framework to do it immediately.

@rogercreagh , would you see any other disadvantages of the swipe to refresh?

**Comment by mueller-ma** [comment]
> Swipe to refresh was there in jtx Board 1.x, actually people were complaining that they didn't wantit because it just triggered unnecessary syncs...

In that case I wouldn't add it back. Sync at start should be sufficient to me.

**Comment by patrickunterwegs** [comment]
So the sync on Resume is now done in #453 
OnResume means that the sync is also triggered when the app is brought to the foreground, no matter if the app was started before.

About the pullRefresh: I have implemented it, it works okay. It might get a bit bothering to see the refresh sign when reaching the top. I'm not sure what to do, keep it or trash it...
I have created this PR here:  Trigger sync through pullRefresh #454 

I have renamed this isssue here to be clear that the sync on resume was done with it and I'll close this issue. It will be part of 2.4.0
